### PR TITLE
[W5.7][W13-4]Anthony Halim

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ooihuiying** [comment]
Review made for W5.7:

Good attempt! Easy to understand and simple code.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Does not change the ordering stored inside the address book.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book alphabetically as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD;
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:21]: Avoid stray line breaks

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:22]: Write descriptive header comments for all public classes/methods.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package seedu.addressbook.data.person;
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/data/person/Contact.java:2]: Write descriptive header comments for all public classes/methods.

File: src/seedu/addressbook/ui/Formatter.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
+package seedu.addressbook.ui;
+
+import java.util.List;
+
+import static seedu.addressbook.common.Messages.*;
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/ui/Formatter.java:5]: Avoid wildcard imports.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 5]
```diff
@@ -34,6 +34,27 @@ public void elementsAreUnique() throws Exception {
...
+
+    @Test
+    public void elementsAreNull() throws Exception {
+
+        // empty list
+        assertNoNull();
+
+        // contain one null object
+        assertContainsNull((Object) null);
+
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:48]: These JUnit tests should not be part of this PR.  Don't mix multiple enhancements in the same PR. Merge previous week's code to the master before branching for a new week's LOs.

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 6]
```diff
@@ -135,4 +142,19 @@ public void getAllPersons() throws Exception {
 
         assertTrue(isIdentical(allPersons, personsToCheck));
     }
+
+    @Test
+    public void nameIsSimilar() throws Exception {
+        Name variationOne = new Name("John K Smith");
+        Name variationTwo = new Name("John K SMIth");
+        Name variationThree = new Name("John Smith");
+//        Name variationFour = new Name("Smith, John K");
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/data/AddressBookTest.java:151]: Avoid commenting code. Delete unused code. You can go back to a previous version, precisely our reason to do version control.

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @SusterDrifter . In particular, good job on,

updating user docs
cleaning up previous code

Suggestions to improve further:
update JUnit tests
update I/O tests
update developer guide

As before, please close the PR after viewing comments.

File: src/seedu/addressbook/ui/Formatter.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * Formats a string as a viewable indexed list item.
+     *
+     * @param visibleIndex visible index for this listing
+     */
+    public static String getIndexedListItem(int visibleIndex, String listItem) {
+        return String.format(MESSAGE_INDEXED_LIST_ITEM, visibleIndex, listItem);
+    }
+}
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/ui/Formatter.java:94]: This change does not seem to be related to the enhancement proposed by this PR. 
Try to limit to one LO per PR.
